applications/GeoMechanicsApplication/tests/cpp_tests/test_utilities/geo_custom_condition.cpp

```diff
@@ -0,0 +1,83 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+#include "custom_utilities/dof_utilities.h"
+#include "geo_custom_condition.h"
+
+namespace Kratos::Testing
+{
+
+void GeoCustomCondition::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix, const ProcessInfo& rCurrentProcessInfo)
+{
+    if (!mStiffnessMatrixSet) {
+        mStiffnessMatrix    = rLeftHandSideMatrix;
+        mStiffnessMatrixSet = true;
+    } else rLeftHandSideMatrix = mStiffnessMatrix;
+}
+
+void GeoCustomCondition::CalculateMassMatrix(MatrixType& rMassMatrix, const ProcessInfo& rCurrentProcessInfo)
+{
+    if (!mMassMatrixSet) {
+        mMassMatrix.resize(rMassMatrix.size1(), rMassMatrix.size2(), false);
+        for (unsigned int i = 0; i < rMassMatrix.size1(); i++) {
+            for (unsigned int j = 0; j < rMassMatrix.size2(); j++) {
+                mMassMatrix(i, j) = rMassMatrix(i, j);
+            }
+        }
```

Comment:
For setting a matrix, I would just use mMassMatrix = rMassMatrix and skip the loops/resizes, as you do for damping/rhs.